lib/Driver/Compilation.cpp

```diff
@@ -458,6 +458,26 @@ namespace driver {
       }
     }
 
+    /// Check to see if a batch-constituent job produced a zero-length
+    /// serialized diagnostics file, which is used to indicate constituents that
+    /// were batched together with a failing constituent but did not,
+    /// themselves, produce any errors.
+    bool batchConstituentJobProducedZeroLengthSerializedDiagnostics(
+      const Job *J) const {
+      auto DiaPath =
+        J->getOutput().getAnyOutputForType(file_types::TY_SerializedDiagnostics);
+      if (!DiaPath.empty()) {
+        if (llvm::sys::fs::is_regular_file(DiaPath)) {
+          uint64_t Size;
+          auto EC = llvm::sys::fs::file_size(DiaPath, Size);
```

Comment:
Along the same handle-errors-first style (I picked up long ago doing OS-oidal coding),
my own style would be moving the error case handling first. Then the whole routine would read better (to my eyes), because it would be series of error tests, each returning false, with the return Size == 0 last. But these are suggestions, not requirements as far as I am concerned.

Reply:
fixed

Reply:
Love how this turned out.